aws/resource_aws_workspaces_directory_test.go

```diff
@@ -0,0 +1,177 @@
+	"github.com/aws/aws-sdk-go/service/workspaces"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccAwsWorkspacesDirectory_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsWorkspacesDirectoryDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccWorkspaceConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsWorkspacesDirectoryExists("aws_workspaces_directory.test"),
+				),
+			},
+			{
+				ResourceName:      "aws_workspaces_directory.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAwsWorkspacesDirectory_subnetIds(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsWorkspacesDirectoryDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccWorkspaceConfig_subnetIds(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsWorkspacesDirectoryExists("aws_workspaces_directory.test"),
+				),
+			},
+			{
+				ResourceName:      "aws_workspaces_directory.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAwsWorkspacesDirectoryDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).workspacesconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_workspaces_directory" {
+			continue
+		}
+
+		resp, err := conn.DescribeWorkspaceDirectories(&workspaces.DescribeWorkspaceDirectoriesInput{
+			DirectoryIds: []*string{aws.String(rs.Primary.ID)},
+		})
+		if err != nil {
+			return err
+		}
+
+		if len(resp.Directories) == 0 {
+			return nil
+		}
+
+		dir := resp.Directories[0]
+		if *dir.State != workspaces.WorkspaceDirectoryStateDeregistering && *dir.State != workspaces.WorkspaceDirectoryStateDeregistered {
+			return fmt.Errorf("directory %q was not deregistered", rs.Primary.ID)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckAwsWorkspacesDirectoryExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("workspaces directory resource is not found: %q", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("workspaces directory resource ID is not set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).workspacesconn
+		resp, err := conn.DescribeWorkspaceDirectories(&workspaces.DescribeWorkspaceDirectoriesInput{
+			DirectoryIds: []*string{aws.String(rs.Primary.ID)},
+		})
+		if err != nil {
+			return err
+		}
+
+		if *resp.Directories[0].DirectoryId == rs.Primary.ID {
+			return nil
+		}
+
+		return fmt.Errorf("workspaces directory %q is not found", rs.Primary.ID)
+	}
+}
+
+func testAccWorkspaceConfig() string {
```

Comment:
If you're not passing any parameters, you can make this and `testAccWorkspaceConfig_subnetIds()` into multi-line string constants